guides/v2.1/test/integration/annotations/magentoDataFixture.md

```diff
@@ -0,0 +1,195 @@
+A data fixture is a PHP script that sets data you want to reuse in your test.
+The script can be defined in a separate file or as a local method in the test case.
+To set up the date fixture, use the `@magentoDataFixture` annotation.
+
+## Format
+
+`@magentoDataFixture` takes an argument that points to the data fixture as a filename a local method.
+
+> Data fixture annotation
+
+```php?start_inline=1
+/**
+ * @magentoDataFixture <script_filename>|<method_name>
+ */
+ ```
+ 
+Here,
+ * `<script_filename>` is a filename of the PHP script
+ * `<method_name>` is a name of the method declared in the current class
+ 
+## Principles
+ 
+1. Do not use direct connection to a database in fixtures.
+It introduces a dependency on the database structure and potentially on the database vendor.
+
+2. Use the application (model) API to implement your data fixtures.
+ 
+## Usage
+ 
+Fixtures can be declared in two ways:
+
+- as a PHP script file that is used by other tests and test cases.
+- as a local method that is used by other tests in the test cases.
+ 
+### Fixture as a separate file
+ 
+Define the fixture in a separate file when you want to reuse it in different test cases.
+ 
+To declare the fixture, use the following convention:
+* a path that is relative to the `dev/tests/integration/<testsuite directory>`
+* containing only forward slashes `/` with no preceding slash. Example: `Magento/Core/Model/Layout/_files/layout_custom_handle.php`
+
+The ITF includes the declared PHP script to your test and executes it during test run.
+
+#### Example
+
+__Use case__: Verify the `getSortOrder()` method that gets the value of `sort_order`.
+
+Create a data fixture as a separate file called `layout_custom_handle.php` that sets `sort_order` to `456`.
+
+> The `layout_custom_handle.php` file content:
+
+```php?start_inline=1
+<?php
+$model = new Magento_Core_Model_Layout_Update();
+$model->setData(array(
+    'handle'     => 'custom',
+    'xml'        => '<layout/>',
+    'sort_order' => 456,
+));
+$model->save();
+```
+
+Declare the fixture in your test case using the `@magentoDataFixture` annotation.
+
+> The test case that uses the data fixture:
+
+```php?start_inline=1
+/**
+ * @magentoDataFixture Magento/Core/Model/Layout/_files/layout_custom_handle.php
+ */
+class Magento_Core_Model_Layout_UpdateTest extends PHPUnit_Framework_TestCase
+{
+    public function testLoad()
+    {
+        /* Load by handle field */
+        $model = new Magento_Core_Model_Layout_Update();
+        $model->load('custom', 'handle');
+        $this->assertEquals(456, $model->getSortOrder());
+    }
+}
+```
+
+### Fixture as a method
+
+A method of the current test case class must be declared as `public` and `static`.
+
+> Fixture Method Usage
+
+```php?start_inline=1
+/**
+ * @magentoDataFixture layoutDataFixture
+ */
+class Magento_Core_Model_Layout_UpdateTest extends PHPUnit_Framework_TestCase
+{
+    private static $_modelId;
+ 
+    public static function layoutDataFixture()
```

Comment:
Please do not use static properties and methods in examples as this is Magento 1 approach and contradicts Magento 2 best practices.

```php

private $modelId;
public function layoutDataFixture()
{
        $model = new \Magento\Widget\Model\Layout\Update();
        $model->setData([
            'handle'     => 'default',
            'xml'        => '<layout/>',
            'sort_order' => 123
        ]);
        $model->save();
        $this->modelId = $model->getId();
}

/**
 * @magentoDataFixture layoutDataFixture
 */
public function testLoad()
{
        /* Load by id */
        $model = new \Magento\Widget\Model\Layout\Update();
        $model->load($this->modelId);
        $this->assertEquals(123, $model->getSortOrder());
}

Reply:
Extended a bit after considering the previous comment about namespace.
Please check if it is correct:

```
namespace Magento\Widget\Model\Layout;

class UpdateTest extends PHPUnit\Framework\TestCase
{
    private $modelId;
    public function layoutDataFixture()
    {
        $model = new Update::class();
        $model->setData([
            'handle'     => 'default',
             'xml'        => '<layout/>',
             'sort_order' => 123
        ]);
        $model->save();
        $this->modelId = $model->getId();
    }
    
    /**
     * @magentoDataFixture layoutDataFixture
     */
    public function testLoad()
    {
        /* Load by id */
        $model = new Update::class();
        $model->load($this->modelId);
        $this->assertEquals(123, $model->getSortOrder());
    }
}
```